naomi-setup.sh

```diff
@@ -7,54 +7,239 @@
-# install dependencies
-sudo apt-get update
-# libssl-dev required to get the python _ssl module working
-sudo apt-get install libssl-dev gettext libncurses5-dev portaudio19-dev libasound2-dev -y
+APT=0
+if command -v apt-get > /dev/null 2>&1 ; then
+    APT=1
+fi
+if [ $APT -eq 1 ]; then
+    echo "Updating apt, preparing to install gettext, portaudio19-dev and libasound2-dev"
+    echo "This operation may require your sudo password"
+    # install dependencies
+    sudo apt-get update
+    # libssl-dev required to get the python _ssl module working
+    sudo apt-get install gettext portaudio19-dev libasound2-dev -y
+else
+    echo "Please ensure that gettext, portaudio and libasound2 are installed on your system"
+fi
+if [ $OPTION = "1" ]; then
+    echo 'VirtualEnv setup'
+    if [ $APT -eq 1 ]; then
+        echo 'Making sure you have the latest python, pip, python3 and pip3 installed on your system'
+        sudo apt-get install python python-pip python3 python3-pip
+    else
+        echo "Please ensure that python, pip, python3 and pip3 are installed on your system"
+    fi
+    pip install --user virtualenvwrapper
+    echo 'sourcing virtualenvwrapper.sh'
+    export VIRTUALENVWRAPPER_VIRTUALENV=~/.local/bin/virtualenv
+    source ~/.local/bin/virtualenvwrapper.sh
+    echo 'checking if Naomi virtualenv exists'
+    workon Naomi > /dev/null 2>&1
+    if [ $? -ne 0 ]; then
+        echo 'Naomi virtualenv does not exist. Creating.'
+        PATH=$PATH:~/.local/bin mkvirtualenv -p python3 Naomi
+        workon Naomi
+    fi
+    if [ $(which pip) = $HOME/.virtualenvs/Naomi/bin/pip ]; then
+        echo 'in the Naomi virtualenv'
+        pip install -r python_requirements.txt
+        deactivate
+    else
+        echo "Something went wrong, not in virtual environment..." >&2
+        exit 1
+    fi
+    # start the naomi setup process
+    echo "#!/bin/bash" > Naomi
+    echo "export VIRTUALENVWRAPPER_VIRTUALENV=~/.local/bin/virtualenv" >> Naomi
+    echo "source ~/.local/bin/virtualenvwrapper.sh" >> Naomi
+    echo "workon Naomi" >> Naomi
+    echo "python $NAOMI_DIR/Naomi.py \$@" >> Naomi
+    echo "deactivate" >> Naomi
+fi
+if [ $OPTION = "2" ]; then
+    if [ $APT -eq "1" ] ; then
+        echo "Making sure you have GnuPG and dirmngr installed"
+        sudo apt-get install libssl-dev libncurses5-dev gnupg dirmngr
+    else
+        echo "Please make sure you have libssl-dev, libncurses5-dev, GnuPG and dirmngr installed"
+    fi
+    # installing python 3.5.3
+    echo 'Installing python 3.5.3 to ~/.naomi/local'
+    mkdir -p ~/.naomi/local
+    cd ~/.naomi/local
+    NAME=Python
+    VERSION=3.5.3
+    VERNAME=$NAME-$VERSION
+    CHKSUM=d8890b84d773cd7059e597dbefa510340de8336ec9b9e9032bf030f19291565a
+    TARFILE=$VERNAME.tgz
+    URL=https://www.python.org/ftp/python/$VERSION/$TARFILE
+    KEYID=3A5CA953F73C700D # Larry Hastings <larry@hastings.org>
+    if [ ! -f $TARFILE ]; then
```

Comment:
Option 3 has failed in a fresh install of RPI x86 image.

I think it has to do something with getting the public key from the server. 
```
    gpg --list-keys $KEYID || gpg --keyserver keys.gnupg.net --recv-keys $KEYID
```

![image](https://user-images.githubusercontent.com/3073252/50806189-e0698080-12b2-11e9-93d8-e3bb3b1f5e71.png)

Reply:
That is very possible, and it seems likely that problems will occur from time to time, just due to the fact that public keyservers go up and down. If you try that again on the same machine, does it work? This does not look like a bug, but the program operating as designed.

Reply:
Thanks for the suggestions. I would prefer to just wrap the instruction lines to a standard 80 char terminal than try to do something tricky with sniffing the terminal width and then finding the closest preceding whitespace to cut at, tabbing over, etc. That gets ridiculously complicated for such a small script, and would probably justify using curses instead of a simple bash script. Again with the colors, that is outside the scope of what I am attempting to accomplish, although if someone wanted to add status coloring later, that would be a fine project. I think you should make these things a separate issue after this PR gets merged.

Thanks for the idea on distro sniffing, but I'm not really trying to create rules for every distro. What I'm trying to do is replace lines like:
Please make sure you have libssl-dev, libncurses5-dev, GnuPG and dirmngr installed
with something that only lists what is actually missing. Like ./configure when it is looking for things, might stop with something like:

libssl development files installed.......no
ssl.h not detected. Exiting.

This tells the user that they need to search for a file named "ssl.h" within the packages available from their package manager and install something that contains this file and also references ssl and dev.

With respect to gpg and dirmngr I should be able to use 'command -v' to tell whether they are installed or not, but with libraries I believe there is a way to do it with ld, but I don't remember. I was hoping someone knew off the top of their head.

I will put a message at the top of the program output, just before the menu, about using sudo, and also give the user an obvious out like "4) Exit without installing".

Thanks again, I know this kind of testing takes a lot of time and effort,
Aaron

Reply:
> That is very possible, and it seems likely that problems will occur from time to time, just due to the fact that public keyservers go up and down. If you try that again on the same machine, does it work? This does not look like a bug, but the program operating as designed.

No it never worked for me since...See my issue #145 for details.

Reply:
> With respect to gpg and dirmngr I should be able to use 'command -v' to tell whether they are installed or not, but with libraries I believe there is a way to do it with ld, but I don't remember. I was hoping someone knew off the top of their head.

I believe you want to use ```ldconfig -p```

o do this in a distro-independent* fashion you can use ldconfig with grep, like this:

```ldconfig -p | grep libjpeg```

If libjpeg is not installed, there will be no output. If it is installed, you will get a line for each version available. 

https://serverfault.com/questions/54736/how-to-check-if-a-library-is-installed

Reply:
Thanks, this helped me get through it and check for the existence of both library headers and executables.